src/ApiException.cs

```diff
@@ -0,0 +1,90 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+
+namespace Laserfiche.Api.Client
+{
+    public partial class ApiException : Exception
+    {
+        /// <summary>
+        /// The API status code.
+        /// </summary>
+        public int StatusCode { get; private set; }
+
+        /// <summary>
+        /// The API error details.
+        /// </summary>
+        public ProblemDetails ProblemDetails { get; private set; }
+
+        /// <summary>
+        /// The API response headers.
+        /// </summary>
+        public IReadOnlyDictionary<string, IEnumerable<string>> Headers { get; private set; }
+
+        public ApiException(string message, int statusCode, IReadOnlyDictionary<string, IEnumerable<string>> headers, ProblemDetails problemDetails, Exception innerException)
+            : base(message, innerException)
+        {
+            StatusCode = statusCode;
+            Headers = headers;
+            ProblemDetails = problemDetails;
+        }
+
+        /// <summary>
+        /// Create an <see cref="ApiException"/>. Will try to deserialize the <paramref name="response"/> to a <see cref="Client.ProblemDetails"/>.
+        /// </summary>
+        /// <param name="statusCode">The response status code.</param>
+        /// <param name="headers">The response headers.</param>
+        /// <param name="response">The response body.</param>
+        /// <param name="jsonSerializerSettings">The json serializer settings used to deserialize the response.</param>
+        /// <param name="innerException">The inner exception.</param>
+        /// <returns></returns>
+        public static ApiException Create(int statusCode, IReadOnlyDictionary<string, IEnumerable<string>> headers, string response, JsonSerializerSettings jsonSerializerSettings, Exception innerException)
```

Comment:
Is there any reason that we use `ApiException.Create` instead of `new ApiException`?

Reply:
I wanted to keep the constructor simple (just set properties). This method does extra work like deserializing the response string, so I thought it made sense to use static factory methods to construct the `ApiException`

Reply:
Got it. Thanks for the explanation.